tutorials/landsat-etm-to-oli-harmonization/script.js

```diff
@@ -0,0 +1,278 @@
+  vAxis: {title: 'NBR'},
+  lineWidth: 6
+});
+print(chartMedianComp);
+
+// ################################################################
+// ### ALTERNATIVE TRANSFORMATION FUNCTIONS ###
+// ################################################################
+
+// Roy et al. (2016) Table 2 provides OLS and RMA regression coefficients
+// to translate ETM+ surface reflectance to OLI surface reflectance and vice
+// versa. The above tutorial demonstrates only ETM+ to OLI transformation by OLS
+// regression. Below are functions for all translation options. Replace above
+// lines: 67 (band name standardization) and 69 (transformation) of the `prepImg`
+// function to apply different transformation options.
+
+// Define OLS and RMA surface regression coefficients.
+var coefficients = {
+  etm2oli_ols: {
+    itcps: ee.Image.constant([0.0003, 0.0088, 0.0061, 0.0412, 0.0254, 0.0172]).multiply(10000),
+    slopes: ee.Image.constant([0.8474, 0.8483, 0.9047, 0.8462, 0.8937, 0.9071])
+  },
+  oli2etm_ols: {
+    itcps: ee.Image.constant([0.0183, 0.0123, 0.0123, 0.0448, 0.0306, 0.0116]).multiply(10000),
+    slopes: ee.Image.constant([0.885, 0.9317, 0.9372, 0.8339, 0.8639, 0.9165])
+  },
+  rma: {
+    itcps: ee.Image.constant([-0.0095, -0.0016, -0.0022, -0.0021, -0.0030, 0.0029]).multiply(10000),
+    slopes: ee.Image.constant([0.9785, 0.9542, 0.9825, 1.0073, 1.0171, 0.9949])
+  }
+};
+
+// Define function to apply OLS ETM+ to OLI transformation.
+function etm2oli_ols(img) {
+  return img.select(['B2', 'B3', 'B4', 'B5', 'B6', 'B7'])
+    .multiply(coefficients.etm2oli_ols.slopes)
+    .add(coefficients.etm2oli_ols.itcps)
+    .round()
+    .toShort()
+    .addBands(img.select('pixel_qa')
+  );
+}
+
+// Define function to apply OLS OLI to ETM+ transformation.
+function oli2etm_ols(img) {
+  return ee.Image(
+    img.select(['B1', 'B2', 'B3', 'B4', 'B5', 'B7'])
+      .multiply(coefficients.oli2etm_ols.slopes)
+      .add(coefficients.oli2etm_ols.itcps)
+      .round()
+      .toShort()
+      .addBands(img.select('pixel_qa'))
+      .copyProperties(img, ['system:time_start'])
+  );
+}
+
+// Define function to apply RMA OLI to ETM+ transformation.
+function oli2etm_rma(img) {
+  return ee.Image(
+    img.select(['B1', 'B2', 'B3', 'B4', 'B5', 'B7'])
+      .subtract(coefficients.rma.itcps)
+      .divide(coefficients.rma.slopes)
+      .round()
+      .toShort()
+      .addBands(img.select('pixel_qa'))
+      .copyProperties(img, ['system:time_start'])
+  );
+}
+
+// Define function to apply RMA ETM+ to OLI transformation.
+function etm2oli_rma(img) {
+  return ee.Image(
+    img.select(['B2', 'B3', 'B4', 'B5', 'B6', 'B7'])
+      .multiply(coefficients.rma.slopes)
+      .add(coefficients.rma.itcps)
+      .round()
+      .toShort()
+      .addBands(img.select('pixel_qa'))
+      .copyProperties(img, ['system:time_start'])
+  );
+}
+
+// Define function to get consistent band names (OLI).
+function oliBands(img) {
+  return ee.Image(ee.Algorithms.If(

```

Comment:
You should really do the prep on the individual collections before merging and about If() as much as possible (since it evaluates both sides essentially doing double work, taking double memory, etc).